testsupport/assertions_use/assertions_test.go

```diff
@@ -0,0 +1,38 @@
+package assertions_use
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	toolchainv1aplha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/assertions"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/assertions/spaceprovisionerconfig"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func Test(t *testing.T) {
+	spcUnderTest := &toolchainv1aplha1.SpaceProvisionerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kachny",
+			Namespace: "default",
+		},
+	}
+
+	scheme := runtime.NewScheme()
+	require.NoError(t, toolchainv1aplha1.AddToScheme(scheme))
+	cl := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(spcUnderTest).Build()
+
+	// use the assertions in a simple immediate call
+	assertions.Test(t, spcUnderTest, spaceprovisionerconfig.That().HasLabel("asdf").HasConditionWithType(toolchainv1aplha1.ConditionReady))
```

Comment:
I would rather stay with the fluent-API approach that we are used to:
```go
spaceprovisionerconfig.AssertThat(t, spcUnderTest).HasLabel("asdf").HasConditionWithType(toolchainv1aplha1.ConditionReady)
```

Reply:
this is different from the `WaitFor` approach. Shouldn't the two be the same (whatever style we pick is fine by me).

Reply:
this style is the same as we use with custom assertions, so I would stay with that.